File: 2032-two-out-of-three/2032-two-out-of-three.go
func twoOutOfThree(nums1 []int, nums2 []int, nums3 []int) []int {
    hashMap := make(map[int][]bool)
    
    iterate := func (nums []int, index int) {
        for i := range nums {
            _, ok := hashMap[nums[i]]
            if !ok {
                hashMap[nums[i]] = []bool {false, false, false}
            }
            hashMap[nums[i]][index] = true
        }
    }
    
    count := func (arr []bool) int {
        count := 0
        for i := range arr {
            if arr[i] {
                count++
            }
        }
        return count
    }
    
    iterate(nums1,0)
    iterate(nums2,1)
    iterate(nums3,2)

    answer := make([]int,0)
    
    for k, v := range hashMap {
        if count(v) >= 2 {
            answer = append(answer,k)
        }
    }
    
    return answer
}

File: 2032-two-out-of-three/2032-two-out-of-three.java
class Solution {
    public List<Integer> twoOutOfThree(int[] nums1, int[] nums2, int[] nums3) {
        List<Integer> answer = new ArrayList<>();
        Map<Integer, List<Integer>> map = new HashMap<>();
        int [][] numsList = new int[][] {nums1, nums2, nums3};
        for (int i = 0; i < numsList.length; i++) {
            updateMap(map, i, numsList[i]);
        }
        for (Map.Entry<Integer, List<Integer>> entry: map.entrySet()) {
            List<Integer> frequencyList = entry.getValue();
            if (appearAtLeastTwice(frequencyList)) {
                answer.add(entry.getKey());
            }
        }
        return answer;
    }

    private void updateMap(Map<Integer, List<Integer>> map, int index, int[] nums) {
        for (int num: nums) {
            List<Integer> frequencyList = map.computeIfAbsent(num, k -> new ArrayList<>(Arrays.asList(0,0,0)));
            frequencyList.set(index, frequencyList.get(index)+1);
        }
    }

    private boolean appearAtLeastTwice(List<Integer> list) {
        boolean appeared = false;
        for (int num: list) {
            if (num == 0) {
                if (appeared) {
                    return false;
                }
                appeared = true;
            }
        }
        return true;
    }
}